### PR TITLE
Wiki thread lighting app

### PR DESCRIPTION
**Pull request by MonicaisHer**
The purpose of this PR is to facilitate easier review and feedback on the content wiki page. The content of this PR will eventually be moved to the wiki page of this repository but will not be merged into the main branch.

This PR depends on https://github.com/canonical/matter-pi-gpio-commander/pull/28 being merged first, as it fixes the issue of Thread commissioning feature on commander snap.

TODO:
 - [ ] Add a link on how to turn on commander's Thread feature

### CODE REVIEW

File: wiki-thread-lighting-app.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+#### Versions Tested in This Guide
+- Matter SDK: [`6b01cb9`](https://github.com/project-chip/connectedhomeip/commit/6b01cb977127eb8547ce66d5b627061dc2dd6c90) (API version: 6)
+- RCP firmware: [`00ac6cd`](https://github.com/openthread/ot-nrf528xx/tree/00ac6cd0137a4f09288b455bf8d7aa72d74062d1) (API version: 6)
+
+#### 1. Form a Thread Network on Machine A
+On Machine A, following this guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 
+and [form a Thread network](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#form-a-thread-network) 
+using the nRF52840 Dongle as the RCP. For details on setting up the RCP, 
+pelase refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:33]:
```suggestion
please refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
```

File: wiki-thread-lighting-app.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+**Option 2**: GPIO Commander Snap
+
+Alternatively, you can use this option to set up the Matter Thread lighting application by installing the Matter Pi GPIO Commander snap. 
+This turns your Raspberry Pi into a Matter Thread lighting device, allowing you to control the connected LED, turning it on or off.
+
+Follow the instructions for installing the matter-pi-gpio-commander snap, 
+provided at [Setup OpenThread Border Router with nRF52840 Dongle](https://github.com/canonical/matter-pi-gpio-commander/wiki/Setup-and-control-a-lighting-device#installation).
+
+TBA: Then, running the GPIO commander snap with Thread feature enabled.
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:70]: Also, need to set the `--thread` flag via the snap option.

File: wiki-thread-lighting-app.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+#### 3. Run OTBR D-Bus Daemon on Machine B
+Check the [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) section
+to run OTBR dbus daemon, which is needed to support lighting app's Thread feature as pointed
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:74]: The dbus daemon runs on the system. What is needed here is to run the OTBR agent. GUI is optional.

The lighting app communicates with OTBR agent via the dbus message bus.

File: wiki-thread-lighting-app.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+#### 1. Form a Thread Network on Machine A
+On Machine A, following this guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 
+and [form a Thread network](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#form-a-thread-network) 
+using the nRF52840 Dongle as the RCP. For details on setting up the RCP, 
+pelase refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
+
+#### 2. Run a Matter Thread Lighting App on Machine B
+There are two options to run a Matter Thread lighting application. 
+
+**Option 1**: Native Build Lighting App
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:38]:
```suggestion
**Option 1**: Native Built Lighting App
```

File: wiki-thread-lighting-app.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+This guide will walk you through the process of building, running, and controlling a Matter Thread lighting application over a Thread network. 
+
+Here are step-by-step instructions:
+
+#### Hardware Setup
+- Machine A
+  - nRF52840 dongle as [Radio Co-Processor (RCP)](https://openthread.io/platforms/co-processor#radio_co-processor_rcp)
+  - Choose one of the following options:
+    - PC running Ubuntu Classic 23.04 OS
+    - Raspberry Pi 4B running Ubuntu Server 22.04 OS or Ubuntu Core 22, and an LED light
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:12]: > Choose one of the following options

A little complicated. Is this implying that Ubuntu Server/Desktop 23.04 on Pi isn't going to work? What about UC22 on a PC?

Maybe the Option 1 and Option 2 referenced in the instructions should also be used here.

```suggestion
  - Option 1: PC running Ubuntu Classic 23.04 OS
  - Option 2: Raspberry Pi 4B running Ubuntu Server 22.04 OS or Ubuntu Core 22, and an LED light
```

File: wiki-thread-lighting-app.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+- Machine B running
+  - OTBR snap
+  - [Chip Tool snap](https://snapcraft.io/chip-tool)
+
+#### Versions Tested in This Guide
+- Matter SDK: [`6b01cb9`](https://github.com/project-chip/connectedhomeip/commit/6b01cb977127eb8547ce66d5b627061dc2dd6c90) (API version: 6)
+- RCP firmware: [`00ac6cd`](https://github.com/openthread/ot-nrf528xx/tree/00ac6cd0137a4f09288b455bf8d7aa72d74062d1) (API version: 6)
+
+#### 1. Form a Thread Network on Machine A
+On Machine A, following this guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap)
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:30]:
```suggestion
On Machine A, following the guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 
```

File: wiki-thread-lighting-app.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+and [form a Thread network](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#form-a-thread-network) 
+using the nRF52840 Dongle as the RCP. For details on setting up the RCP, 
+pelase refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
+
+#### 2. Run a Matter Thread Lighting App on Machine B
+There are two options to run a Matter Thread lighting application. 
+
+**Option 1**: Native Build Lighting App
+
+You can choose this option to build and run the Matter Thread lighting app from the Matter SDK in the native shell. With this option, you can observe the lighting status (on/off) from the logs.
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:40]:
```suggestion
You can choose this option to build and run the [Matter Thread lighting app example](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux) natively. With this option, you can observe the lighting status (on/off) from the logs.
```

**Review by farshidtz — CHANGES_REQUESTED** [review]
The Option 1 / Option 2 makes the how-to a little hard to follow, but it is nevertheless a good guide given the lack of any upstream documentation.

File: wiki-thread-lighting-app.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+This guide will walk you through the process of building, running, and controlling a Matter Thread lighting application over a Thread network. 
+
+Here are step-by-step instructions:
+
+#### Hardware Setup
+- Machine A
+  - nRF52840 dongle as [Radio Co-Processor (RCP)](https://openthread.io/platforms/co-processor#radio_co-processor_rcp)
+  - Choose one of the following options:
+    - PC running Ubuntu Classic 23.04 OS
```
[the remark this answers is not in the document]
Comment (MonicaisHer) [reply wiki-thread-lighting-app.md]: I was referring to classic Ubuntu OS variants, such as Ubuntu Server and Ubuntu Desktop. It has now been updated to the one I tested, which is Ubuntu Desktop.

File: wiki-thread-lighting-app.md (hunk 2 above)
Comment (MonicaisHer) [reply wiki-thread-lighting-app.md:70]: Indeed, this information regarding the --thread flag could be included on the GPIO Commander Snap's side. And a link could be simply added here to direct users to it.

File: wiki-thread-lighting-app.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (farshidtz) [reply wiki-thread-lighting-app.md]: You could say "Ubuntu Server/Desktop". There is no distro named "Ubuntu Classic"; Canonical refers to non-Core Ubuntu variants as the [classic](https://www.merriam-webster.com/dictionary/classic) Ubuntu.

File: wiki-thread-lighting-app.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+#### 3. Run OpenThread Border Router Agent on Machine B
+Check the [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) section
+to run OTBR agent. The lighting app communicates with OTBR agent via the dbus message bus. This is necessary to support the lighting app's Thread feature, as explained
+[here](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux#commandline-arguments).
+
+> Note: Only the OTBR agent is required, and the GUI (openthread-border-router.otbr-web) is optional. The OTBR agent can be started by running the following command:
+> ```bash
+> sudo snap start openthread-border-router.otbr-agent
+> ```
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:80]: The link on line 73 can be changed to a link to the README for installation and configuration. The rest of the needed operations can be documented here:
1. Install and config
2. Plugging the dongle
3. Starting the agent (for details, refer [here](https://github.com/canonical/openthread-border-router-snap#run))

**Review by farshidtz — CHANGES_REQUESTED** [review]
I suggest removing Option 2 all-together to simplify this how to. The Option 2 mode can be its own wiki page maintained at https://github.com/canonical/matter-pi-gpio-commander/wiki

File: wiki-thread-lighting-app.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+  - [Chip Tool snap](https://snapcraft.io/chip-tool)
+
+#### Versions Tested in This Guide
+- Matter SDK: [`6b01cb9`](https://github.com/project-chip/connectedhomeip/commit/6b01cb977127eb8547ce66d5b627061dc2dd6c90) (API version: 6)
+- RCP firmware: [`00ac6cd`](https://github.com/openthread/ot-nrf528xx/tree/00ac6cd0137a4f09288b455bf8d7aa72d74062d1) (API version: 6)
+
+#### 1. Form a Thread Network on Machine A
+On Machine A, following the guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 
+and [form a Thread network](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#form-a-thread-network) 
+using the nRF52840 Dongle as the RCP. For details on setting up the RCP,
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:30]:
```suggestion
using the nRF52840 Dongle as the RCP. For details on setting up the OpenThread RCP, 
```

File: wiki-thread-lighting-app.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+#### Versions Tested in This Guide
+- Matter SDK: [`6b01cb9`](https://github.com/project-chip/connectedhomeip/commit/6b01cb977127eb8547ce66d5b627061dc2dd6c90) (API version: 6)
+- RCP firmware: [`00ac6cd`](https://github.com/openthread/ot-nrf528xx/tree/00ac6cd0137a4f09288b455bf8d7aa72d74062d1) (API version: 6)
+
+#### 1. Form a Thread Network on Machine A
+On Machine A, following the guide to [install and configure the OTBR snap](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#install-and-configure-the-otbr-snap) 
+and [form a Thread network](https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-control-a-Matter-Thread-device-via-the-OTBR-Snap#form-a-thread-network) 
+using the nRF52840 Dongle as the RCP. For details on setting up the RCP, 
+please refer to [Build and flash RCP firmware on nRF52480 dongle](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:31]: Redundant text.
```suggestion
please refer [here](https://github.com/canonical/openthread-border-router-snap/wiki/Setup-OpenThread-Border-Router-with-nRF52840-Dongle#build-and-flash-rcp-firmware-on-nrf52480-dongle).
```

File: wiki-thread-lighting-app.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+
+#### 2. Run a Matter Thread Lighting App on Machine B
+
+Build and run the [Matter Thread lighting app example](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux) natively. By doing so, you will be able to observe the lighting status (on/off) from the logs later.
```
Comment (farshidtz) [inline wiki-thread-lighting-app.md:35]:
```suggestion
Build and run the [Matter lighting app example](https://github.com/project-chip/connectedhomeip/tree/6b01cb977127eb8547ce66d5b627061dc2dd6c90/examples/lighting-app/linux) natively. By doing so, you will be able to observe the lighting status (on/off) from the logs later.
```

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: wiki-thread-lighting-app.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+- Machine A running
+  - [OpenThread Border Router (OTBR) snap](https://snapcraft.io/openthread-border-router)
+  - Matter Thread ligting App
+- Machine B running
+  - OTBR snap
+  - [Chip Tool snap](https://snapcraft.io/chip-tool)
+
+#### Versions Tested in This Guide
+- Matter SDK: [`6b01cb9`](https://github.com/project-chip/connectedhomeip/commit/6b01cb977127eb8547ce66d5b627061dc2dd6c90) (API version: 6)
+- RCP firmware: [`00ac6cd`](https://github.com/openthread/ot-nrf528xx/tree/00ac6cd0137a4f09288b455bf8d7aa72d74062d1) (API version: 6)
```
[the remark this answers is not in the document]
Comment (MonicaisHer) [reply wiki-thread-lighting-app.md]: The version should actually refers to the openthread SDK, not the Matter SDK. Reference links have been added in the latest commit.

**Review by farshidtz — APPROVED** [review]

**Comment by farshidtz** [comment]
Thanks. Ready to move over to wiki.

**Comment by MonicaisHer** [comment]
The content has been moved to wiki now: https://github.com/canonical/openthread-border-router-snap/wiki/Commission-and-Control-a-Matter-Thread-Lighting-Device-via-the-OTBR-Snap
